packages/vitest/src/node/reporters/base.ts

```diff
@@ -198,9 +200,31 @@ export abstract class BaseReporter implements Reporter {
     const header = c.gray(log.type + c.dim(` | ${task ? getFullName(task, c.dim(' > ')) : log.taskId !== UNKNOWN_TEST_ID ? log.taskId : 'unknown test'}`))
 
     const output = log.type === 'stdout' ? this.ctx.logger.outputStream : this.ctx.logger.errorStream
+    const write = (msg: string) => (output as any).write(msg)
+
+    write(`${header}\n${log.content}`)
+
+    if (log.origin) {
+      // browser logs don't have an extra end of line at the end like Node.js does
+      if (log.browser)
+        write('\n')
+      const project = log.taskId
+        ? this.ctx.getProjectByTaskId(log.taskId)
+        : this.ctx.getCoreWorkspaceProject()
+      const stack = parseStacktrace(log.origin, {
+        getSourceMap: file => project.getBrowserSourceMapModuleById(file),
+        frameFilter: project.config.onStackTrace,
+      })
+      const highlight = task ? stack.find(i => i.file === task.file.filepath) : null
+      for (const frame of stack) {
+        const color = frame === highlight ? c.cyan : c.gray
+        const path = relative(project.config.root, frame.file)
+
+        write(color(` ${c.dim(F_POINTER)} ${[frame.method, `${path}:${c.dim(`${frame.line}:${frame.column}`)}`].filter(Boolean).join(' ')}\n`))
+      }
+    }
 
-    // @ts-expect-error -- write() method has different signature on the union type
-    output.write(`${header}\n${log.content}\n`)
+    write('\n')
```

Comment:
This kind of `write` usage seems to cause `Ineffective mark-compacts near heap limit` 🤔  (Or something else in this PR...)